solver.go

```diff
@@ -347,29 +367,32 @@ func (s *solver) NewSession(ctx context.Context, cln *client.Client, p progress.
 		allSessions[attach] = bkSess
 	}
 
-	resolver := newResolver(ctx, cln, p)
-	releasers = append(releasers, resolver.stop)
-
-	eg, ctx := errgroup.WithContext(ctx)
-	releasers = append(releasers, eg.Wait)
 	waiters := []chan struct{}{}
+
 	for _, bkSess := range allSessions {
 		bkSess := bkSess
-		done := make(chan struct{})
-		waiters = append(waiters, done)
-		eg.Go(func() error {
-			return bkSess.Run(ctx, func(ctx context.Context, proto string, meta map[string][]string) (net.Conn, error) {
-				defer close(done)
-				return cln.Dialer()(ctx, proto, meta)
+		alive := make(chan struct{})
+		waiters = append(waiters, alive)
+		go func() {
```

Comment:
moved this out of the errgroup since the eg.Wait was part of the cleanup `releasers` and we should close the sessions after the releasers in lifo order.  I confirmed that the session `Run` will exit on when session.Close returns, also added goleak test to verify.